### PR TITLE
Remove RCT2 interop

### PR DESCRIPTION
**Pull request by IntelOrca**
Removes all `!NO_RCT2` blocks, defines etc.
Testpaint still works, although it is on the verge of breaking until we get RCT2 versions of the structs testpaint uses.

### CODE REVIEW

File: src/openrct2/ride/Vehicle.h [hunk 1]
```diff
@@ -21,12 +21,10 @@
 #include "../world/map.h"
 
 #pragma pack(push, 1)
-/* size: 0x2 */
```
Comment (janisozaur) [inline src/openrct2/ride/Vehicle.h]: are these not stored in sv6 or other files?

Comment (IntelOrca) [reply src/openrct2/ride/Vehicle.h]: They are, but all are loading is now done by reading each field individually.

File: test/testpaint/PaintIntercept.cpp [hunk 2]
```diff
@@ -273,7 +273,7 @@ namespace PaintIntercept {
     }
 
     static uint8 InterceptPaintFull(uint8 function, registers *regs) {
-        if ((regs->ebp & 0x03) != get_current_rotation()) {
+        if ((regs->ebp & 0x03) != RCT2_GLOBAL(0x0141E9E0, uint8)) {
```
Comment (janisozaur) [inline test/testpaint/PaintIntercept.cpp]: put them in `TestPaint.hpp` maybe?

File: src/openrct2/CMakeLists.txt [hunk 3]
```diff
@@ -261,15 +185,6 @@ endif ()
...
     add_definitions(-D__ENABLE_LIGHTFX__)
 endif ()
-if (DISABLE_RCT2)
-    add_definitions(-DNO_RCT2)
-endif ()
-
-if (NOT DISABLE_RCT2)
-    # Disable optimizations for addresses.c for all compilers, to allow optimized
-    # builds without need for -fno-omit-frame-pointer
-    set_source_files_properties(${CMAKE_CURRENT_LIST_DIR}/rct2/addresses.c PROPERTIES COMPILE_FLAGS -O0)
```
Comment (janisozaur) [inline src/openrct2/CMakeLists.txt]: This is not reflected in tespaint's cmake

File: test/testpaint/TestPaint.cpp [hunk 4]
```diff
@@ -59,10 +59,18 @@ namespace TestPaint
         rideEntry.vehicles[0] = vehicleEntry;
 
         gRideList[0] = ride;
+        RCT2_ADDRESS(0x013628F8, Ride)[0] = ride;
```
Comment (Gymnasiast) [inline test/testpaint/TestPaint.cpp]: Can we have these addressed named?

File: test/testpaint/TestTrack.cpp [hunk 5]
```diff
@@ -166,6 +166,9 @@ class EntranceStyleFilter : public ITestTrackFilter {
                          Ride *ride, rct_ride_entry *rideEntry
     ) override {
         ride->entrance_style = variant;
+
+        Ride * rct2ride = RCT2_ADDRESS(0x013628F8, Ride);
```
Comment (Gymnasiast) [inline test/testpaint/TestTrack.cpp]: Perhaps you could use `rct2_ride` in TestPaint?

Comment (IntelOrca) [reply test/testpaint/TestTrack.cpp]: Will do.

**Comment by Gymnasiast** [comment]
I fixed the Xcode project. I also removed many "recovered references", which are just files that Xcode could not find anymore, because those are ported to C++.

**Comment by Gymnasiast** [comment]
Generally, it looks reasonable to me.

Have you tried running Testpaint before and after too see if the results match?

File: src/openrct2/ride/Vehicle.h [hunk 6]
```diff
@@ -21,12 +21,10 @@
 #include "../world/map.h"
 
 #pragma pack(push, 1)
```
Comment (janisozaur) [inline src/openrct2/ride/Vehicle.h]: This is not needed now then, based on your comment below. Furthermore, we only ever checked for size of `rct_ride_entry_vehicle` in with-RCT2 builds

File: src/openrct2/ride/ride.h [hunk 7]
```diff
@@ -125,14 +125,9 @@ typedef struct rct_ride_entry {
     uint8 category[RCT2_MAX_CATEGORIES_PER_RIDE];       // 0x1BE
     uint8 shop_item;                                    // 0x1C0
     uint8 shop_item_secondary;                          // 0x1C1
-#if defined(NO_RCT2)
     rct_string_id capacity;
     rct_string_id vehicleName;
-#endif
 } rct_ride_entry;
```
Comment (janisozaur) [inline src/openrct2/ride/ride.h]: If this structure no longer needs its size checked, remove it out of packing pragmas

File: test/testpaint/TestPaint.cpp [hunk 8]
```diff
@@ -47,22 +47,37 @@ namespace TestPaint
...
+            gRideList[0] = ride;
+            gRideEntries[0] = &rideEntry;
+        }
+        {
+            rct2_ride ride;
+            ride.entrance_style = RIDE_ENTRANCE_STYLE_PLAIN;
+            rct_ride_entry rideEntry = {0};
+            rct_ride_entry_vehicle vehicleEntry { 0 };
+            vehicleEntry.base_image_id = 0x70000;
+            rideEntry.vehicles[0] = vehicleEntry;
```
Comment (janisozaur) [inline test/testpaint/TestPaint.cpp]: `rideEntry` is assigned but unused, and when removed, so is `vehicleEntry`.

File: test/testpaint/TestPaint.cpp [hunk 9]
```diff
@@ -47,22 +47,37 @@ namespace TestPaint
...
-        gRideEntries[0] = &rideEntry;
+        {
+            Ride ride = {0};
+            ride.entrance_style = RIDE_ENTRANCE_STYLE_PLAIN;
+            rct_ride_entry rideEntry = {0};
+            rct_ride_entry_vehicle vehicleEntry { 0 };
+            vehicleEntry.base_image_id = 0x70000;
+            rideEntry.vehicles[0] = vehicleEntry;
+            gRideList[0] = ride;
+            gRideEntries[0] = &rideEntry;
```
Comment (janisozaur) [inline test/testpaint/TestPaint.cpp]: Doesn't look right? You're taking address of stack variable here.

File: src/openrct2/ride/ride.h (hunk 7 above)
Comment (IntelOrca) [reply src/openrct2/ride/ride.h]: It does still need to be packed as it is still used in testpaint.

File: test/testpaint/TestPaint.cpp [hunk 10]
```diff
@@ -47,22 +47,37 @@ namespace TestPaint
...
+        }
+        {
+            rct2_ride ride;
```
Comment (janisozaur) [inline test/testpaint/TestPaint.cpp]: this is left uninitialised, other than `entrance_style` in line below

File: test/testpaint/addresses.h [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+ * P = Parity flag
+ * All other bits are undefined.
+ */
+sint32 RCT2_CALLPROC_X(sint32 address, sint32 _eax, sint32 _ebx, sint32 _ecx, sint32 _edx, sint32 _esi, sint32 _edi, sint32 _ebp);
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif
```
Comment (janisozaur) [inline test/testpaint/addresses.h]: missing endline

File: test/testpaint/addresses.h [hunk 12]
```diff
@@ -0,0 +1,61 @@
+
```
Comment (janisozaur) [inline test/testpaint/addresses.h]: spurious line

File: src/openrct2/ride/Vehicle.h [hunk 13]
```diff
@@ -20,12 +20,12 @@
 #include "../common.h"
 #include "../world/map.h"
 
-#pragma pack(push, 1)
 typedef struct rct_vehicle_colour {
     uint8 body_colour;
     uint8 trim_colour;
 } rct_vehicle_colour;
 
+#pragma pack(push, 1)
```
Comment (janisozaur) [inline src/openrct2/ride/Vehicle.h]: just remove it completely, `rct_ride_entry_vehicle` doesn't have its size checked either. (please verify it still loads and ride ratings test works)

Comment (IntelOrca) [reply src/openrct2/ride/Vehicle.h]: If I get rid of it, it crashes.

**Pull request merged**